### DISCUSSION TITLE
Prevent User to login into an active Session twice and kick out the current

### DISCUSSION THREAD

**Discussion by itxworks** [neutrinolabs/xrdp#2616] [Q&A]
Hello,

maybe I missed it somewhere - is there a easy way to prevent the same user to login over an active session on the same server ...

Thank you.

**Comment by matt335672** [comment]
At the moment, no.

With the existing architecture, when a user connects, xrdp-sesman has no way to communicate with the xrdp process in an existing session to tell it to drop out.

This has caused other problems when firewalls are involved (#1945 and others).

We're working towards a solution - see #1961. The initial deployment of this won't support this functionality but it will at least be possible.

**Comment by itxworks** [comment]
OK - would it be possible to make a check for an active check in the startwm.sh ...

Thank you

**Comment by matt335672** [comment]
Sadly not easily. There's no way to determine which xrdp process corresponds to which user. On non-systemd systems, users can each have multiple session active.

**Comment by itxworks** [comment]
Right ... I tried to use something like this https://github.com/evanlinde/xrdp-who

....

Thank you again ...
